tools/codegen/model.py

```diff
@@ -1496,3 +1503,40 @@ def parse_returns(return_decl: str) -> Tuple[Return, ...]:
     if return_decl[0] == '(' and return_decl[-1] == ')':
         return_decl = return_decl[1:-1]
     return tuple(Return.parse(arg) for arg in return_decl.split(', '))
+
+@dataclass(frozen=True)
+class PrecomputedElement:
+    name: str
+    ty: str
```

Comment:
This is one of those places where telling people in a comment that this is a C++ type specifically is really, really helpful semantic information for downstream users. I might go as far to say you should call this `cpp_ty` and not just ty. (This is moot if we decide to do JIT schema types insetad)